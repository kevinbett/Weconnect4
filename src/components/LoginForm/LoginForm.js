import React from 'react';
import {Link} from 'react-router-dom';
import './style.css';


class LoginForm extends React.Component {
  
        state = {
        message: undefined
      }

    handleUserLogin = async (event) => {
      event.preventDefault();
      const email = event.target.emailid.value;
      const password = event.target.password.value;
      const apiData = await fetch('https://weconnect4-heroku.herokuapp.com/login', {
        method: "POST",
        headers: {'Accept': 'application/json','Content-Type': 'application/json'},
        body: JSON.stringify({
          "email": email,
          "password": password

        })
  });

  const res = await apiData.json();
  console.log(res)
  
  if (res.auth_token) {
    window.localStorage.setItem("authToken", res.auth_token)
    window.localStorage.setItem("user_id", res.user_id)
    window.location.assign('/viewbusiness')
  } else {
    window.localStorage.setItem("rMessage", res.message)
  }
  this.setState({message: res.message, error: res.error});
}

  componentWillUnmount () {
    window.localStorage.removeItem('rMessage')
  }

  render() {
    return (
    <div id="main-registration-container">
     <div id="register">
        <h3>User Login</h3>
        <form onSubmit= {this.handleUserLogin} >
        <p className="errorMsg">{window.localStorage.getItem("rMessage")}</p>
        <label>Email ID:</label>
        <input type="text" name="emailid" />
        <label>Password</label>
        <input type="password" name="password" />

        <input type="submit" className="button"  value="Login"/>
        </form>
        <Link className="pass-reset" to="/resetpassword">Forgot password?..</Link>
        
    </div>
</div>

      );
  }


}


export default LoginForm;