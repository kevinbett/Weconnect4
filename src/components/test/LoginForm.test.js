import LoginForm from "../LoginForm/LoginForm"
import { shallow } from "enzyme"

describe('login page renders',() => { 
    const wrapper = shallow(<LoginForm/>);
    it ('populates the the login form', () => { 
        expect(shallow(<LoginForm/>)).toMatchSnapshot()
    });

} )
