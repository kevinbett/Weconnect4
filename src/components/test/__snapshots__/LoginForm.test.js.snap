// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login page renders populates the the login form 1`] = `
<div
  id="main-registration-container"
>
  <div
    id="register"
  >
    <h3>
      User Login
    </h3>
    <form
      onSubmit={[Function]}
    >
      <label>
        Email ID:
      </label>
      <input
        name="emailid"
        type="text"
      />
      <label>
        Password
      </label>
      <input
        name="password"
        type="password"
      />
      <input
        className="button"
        type="submit"
        value="Login"
      />
    </form>
    <Link
      className="pass-reset"
      replace={false}
      to="/resetpassword"
    >
      Forgot password?..
    </Link>
  </div>
</div>
`;
